tools/minicluster/minicluster.go

```diff
@@ -96,15 +97,16 @@ func main() {
 		go itx.StartServer(context.Background(), svrs[i], probe.New(7788), configs[i])
 	}
 
+	// target address for grpc connection. Default is "127.0.0.1:14014"
+	grpcAddr := "127.0.0.1:14014"
+	var conn *grpc.ClientConn
 	if err := testutil.WaitUntil(10*time.Millisecond, 2*time.Second, func() (bool, error) {
-		return svrs[0].ChainService(uint32(1)).Explorer().Port() == 14004, nil
+		conn, err = grpc.Dial(grpcAddr, grpc.WithInsecure())
```

Comment:
Please try to wait until the server is ready to establish the connection. You could probe the server readiness and block on this instead